modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import {ClayRadio, ClayRadioGroup} from '@clayui/form';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayModal, {useModal} from '@clayui/modal';
+import EditTagsContext from './EditTagsContext.es';
+import PropTypes from 'prop-types';
+import React, {
+	useCallback,
+	useContext,
+	useEffect,
+	useRef,
+	useState
+} from 'react';
+import {fetch} from 'frontend-js-web';
+import {AssetTagsSelector} from 'asset-taglib';
+
+const EditTagsModal = ({
+	fileEntries,
+	folderId,
+	groupIds,
+	pathModule,
+	repositoryId,
+	selectAll = false,
+	urlSelection = '/bulk/v1.0/bulk-selection',
+	urlTags = '/bulk/v1.0/keywords/common',
+	urlUpdateTags = '/bulk/v1.0/keywords/batch',
+	onModalClose = () => {}
+}) => {
+	const {namespace} = useContext(EditTagsContext);
+
+	const [append, setAppend] = useState(true);
+	const [bulkSelection, setBulkSelection] = useState();
+	const [description, setDescription] = useState('');
+	const [loading, setLoading] = useState(true);
+	const [multiple, setMultiple] = useState(false);
+	const [initialSelectedItems, setInitialSelectedItems] = useState([]);
+	const [selectedItems, setSelectedItems] = useState([]);
+	const [selectedRadioGroupValue, setSelectedRadioGroupValue] = useState(
+		'add'
+	);
+
+	const {observer, onClose} = useModal({
+		onClose: onModalClose
+	});
+
+	useEffect(() => {
+		setBulkSelection(getBulkSelection());
+	}, [fileEntries, folderId, getBulkSelection, repositoryId, selectAll]);
```

Comment:
We have the same problem here that we had on [an earlier version of this pull](https://github.com/wincent/liferay-portal/pull/60) which is that you are not supposed to access `const` bindings when you are in the "[temporal dead zone](https://2ality.com/2015/10/why-tdz.html)". For example, `getBulkSelection` is accessed here, but is only defined on line 116. I think we are only getting away with it because Babel transforms it into a `var`, but you can see that we'll get a `ReferenceError` if we ever ship this code untranspiled to a modern browser:

```javascript
function x() {
    console.log(y);
    const y = 1;
}

x(); // Thrown: ReferenceError: y is not defined
```

So, we shouldn't do that. Have a look through and see if there are any other `const` bindings (`let` too would be a problem in the temporal dead zone) that are being accessed before their declarations.